llvm/tools/llvm-readobj/ELFDumper.cpp

```diff
@@ -5513,41 +5513,42 @@ static AMDNote getAMDNote(uint32_t NoteType, ArrayRef<uint8_t> Desc) {
 struct AMDGPUNote {
   std::string Type;
   std::string Value;
+  bool IsError;
 };
 
 template <typename ELFT>
 static AMDGPUNote getAMDGPUNote(uint32_t NoteType, ArrayRef<uint8_t> Desc) {
```

Comment:
Maybe this should return Expected<AMDGPUNote> instead of adding the extra boolean field?

Reply:
This isn't really a "I'm returning either an Error or a valid AMDGPUNote" situation. In the error case, the Error object gets written into the std::string fields, I just wanted to indicate to the caller that there was an error pretty-printing so it would still dump the binary blob, which seems useful to include.

Reply:
This should still be possible if you return an Expected<> object? We can just print the error string from the bad Expected<> and then continue to the raw dumping?

Reply:
Something like
```c++
      const Expected<AMDGPUNote> N = getAMDGPUNote<ELFT>(Type, Descriptor);
      if (!N) {
        OS << "    "AMDGPU Metadata:\n        " <<toString(N.takeError())) << '\n';
        // Fallthrough to printing the description data blob on error.
      } else {
        OS << "    " << N->Type << ":\n        " << N->Value << '\n';
        return Error::success();
      }
```